metrics/models.py

```diff
@@ -0,0 +1,64 @@
+from django.db import models
+
+
+class Sensor(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    # example --> 28-3c01d607df32
+    uuid = models.CharField(max_length=100)
+
+    # example --> DS18B20
+    sensor_model = models.CharField(max_length=255)
+
+    # example --> sensor_1
+    title = models.CharField(max_length=255)
+
+    # example --> This sensor used in room number 2
+    description = models.TextField(max_length=10000)
+
+    # example --> 1 (temperature)
+    type = models.ForeignKey(to="metrics.SensorType", on_delete=models.CASCADE)
+
+    # example --> 2 (celsius)
+    unit = models.ForeignKey(to="metrics.Unit", on_delete=models.CASCADE)
+
+    def __str__(self):
+        return self.uuid
+
+
+class SensorType(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    # example --> temperature
+    name = models.CharField(max_length=64)
+
+    def __str__(self):
+        return self.name
+
+
+class Unit(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    # example --> celsius
+    name = models.CharField(max_length=64)
+
+    def __str__(self):
+        return self.name
+
+
+class   Log(models.Model):
+    # example --> 1 (ds18b20)
+    sensor = models.ForeignKey(to="metrics.Sensor", on_delete=models.CASCADE)
+
+    created_at_by_server = models.DateTimeField(auto_now_add=True)
+    updated_at_by_server = models.DateTimeField(auto_now=True)
```

Comment:
There are two problems with these names:

- First, the logic behind `created_at` is when seeing as key-value has meaning for, example "created_at: 2020-10-4" and `created_at_by_server` doesn't have any meaning in key-value form.
- Second, when you have to name fields like this it can be a clue to more serious problems in logics and system design.

Reply:
fixed